contracts/cw-ics721-bridge/src/contract.rs

```diff
@@ -356,9 +361,53 @@ pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
         QueryMsg::Pauser {} => to_binary(&PO.query_pauser(deps.storage)?),
         QueryMsg::Paused {} => to_binary(&PO.query_paused(deps.storage)?),
         QueryMsg::Proxy {} => to_binary(&PROXY.load(deps.storage)?),
+        QueryMsg::Cw721CodeId {} => to_binary(&query_cw721_code_id(deps)?),
+        QueryMsg::ClassIdToNftContract {} => to_binary(&query_class_id_to_nft_contract(deps)?),
+        QueryMsg::OutgoingClassTokenToChannel {} => to_binary(&query_class_token_to_channel(
+            deps,
+            OUTGOING_CLASS_TOKEN_TO_CHANNEL,
+        )?),
+        QueryMsg::IncomingClassTokenToChannel {} => to_binary(&query_class_token_to_channel(
+            deps,
+            INCOMING_CLASS_TOKEN_TO_CHANNEL,
+        )?),
     }
 }
 
+fn query_cw721_code_id(deps: Deps) -> StdResult<u64> {
+    CW721_CODE_ID.load(deps.storage)
+}
+
+fn query_class_id_to_nft_contract(deps: Deps) -> StdResult<Vec<ClassIdToNftContractResponse>> {
+    CLASS_ID_TO_NFT_CONTRACT
+        .range(deps.storage, None, None, Order::Ascending)
+        .map(|p| {
+            let (class_id, nft_contract) = p?;
+            Ok(ClassIdToNftContractResponse {
+                class_id,
+                nft_contract,
+            })
+        })
+        .collect::<StdResult<Vec<ClassIdToNftContractResponse>>>()
+}
+
+fn query_class_token_to_channel(
+    deps: Deps,
+    class_token_to_channel: Map<(ClassId, TokenId), String>,
+) -> StdResult<Vec<ClassTokenToChannelResponse>> {
+    class_token_to_channel
+        .range(deps.storage, None, None, Order::Ascending)
+        .map(|p| {
+            let ((class_id, token_id), channel) = p?;
+            Ok(ClassTokenToChannelResponse {
+                class_id,
+                token_id,
+                channel,
+            })
+        })
+        .collect::<StdResult<Vec<ClassTokenToChannelResponse>>>()
+}
```

Comment:
I like the abstraction here, my concern is that this query is not paginated so once a couple NFTs have been transferred it will run out of gas. Could we add some pagination to this? For example, [this function](https://github.com/DA0-DA0/dao-contracts/blob/main/packages/cw-paginate/src/lib.rs#L10-L33) might be useful.

Reply:
Good point. Will add pagination for all quieries here. Thx.

Reply:
Pagination added. Pls check.